src/Fluid/Flu_FixUp_Restrict.cpp

```diff
@@ -343,6 +343,24 @@ void Flu_FixUp_Restrict( const int FaLv, const int SonFluSg, const int FaFluSg,
                                         amr->patch[FaFluSg][FaLv][FaPID]->fluid[ENGY][k][j][i],
                                         MIN_EINT, Emag );
 #        endif // #ifdef DUAL_ENERGY ... else ...
+
+//       fix temperature initial guess
+#        ifdef TEMP_IG
+         real Passive[NCOMP_PASSIVE];
+
+         for (int N_PASSIVE=0; N_PASSIVE<NCOMP_PASSIVE; N_PASSIVE++)
+            Passive[N_PASSIVE] = amr->patch[FaFluSg][FaLv][FaPID]->fluid[NCOMP_FLUID+N_PASSIVE][k][j][i];
+
+         amr->patch[FaFluSg][FaLv][FaPID]->fluid[TEMP_IG][k][j][i]
+            = Hydro_Con2Temp( amr->patch[FaFluSg][FaLv][FaPID]->fluid[DENS][k][j][i],
+                              amr->patch[FaFluSg][FaLv][FaPID]->fluid[MOMX][k][j][i],
+                              amr->patch[FaFluSg][FaLv][FaPID]->fluid[MOMY][k][j][i],
+                              amr->patch[FaFluSg][FaLv][FaPID]->fluid[MOMZ][k][j][i],
+                              amr->patch[FaFluSg][FaLv][FaPID]->fluid[ENGY][k][j][i],
+                              Passive,
+                              false, NULL_REAL, Emag, EoS_DensEint2Temp_CPUPtr,
```

Comment:
Replace `false` by defining `const bool CheckMinTemp_No = false;`

Reply:
I have revised this.